src/zenml/integrations/aws/orchestrators/sagemaker_orchestrator.py

```diff
@@ -0,0 +1,170 @@
+            return os.environ[ENV_ZENML_SAGEMAKER_RUN_ID]
+        except KeyError:
+            raise RuntimeError(
+                "Unable to read run id from environment variable "
+                f"{ENV_ZENML_SAGEMAKER_RUN_ID}."
+            )
+
+    def prepare_pipeline_deployment(
+        self, deployment: "PipelineDeployment", stack: "Stack"
+    ) -> None:
+        """Build a Docker image and push it to the container registry.
+
+        Args:
+            deployment: The pipeline deployment configuration.
+            stack: The stack on which the pipeline will be deployed.
+        """
+        docker_image_builder = PipelineDockerImageBuilder()
+        repo_digest = docker_image_builder.build_and_push_docker_image(
+            deployment=deployment, stack=stack
+        )
+        deployment.add_extra(ORCHESTRATOR_DOCKER_IMAGE_KEY, repo_digest)
+
+    @property
+    def settings_class(self) -> Optional[Type["BaseSettings"]]:
+        """Settings class for the Sagemaker orchestrator.
+
+        Returns:
+            The settings class.
+        """
+        return SagemakerOrchestratorSettings
+
+    def prepare_or_run_pipeline(
+        self, deployment: "PipelineDeployment", stack: "Stack"
+    ) -> Any:
+        """Prepares or runs a pipeline on Sagemaker.
+
+        Args:
+            deployment: The deployment to prepare or run.
+            stack: The stack to run on.
+
+        Returns:
+            The result of the pipeline run.
+        """
+        if deployment.schedule:
+            logger.warning(
+                "The Sagemaker Orchestrator currently does not support the "
+                "use of schedules. The `schedule` will be ignored "
+                "and the pipeline will be run immediately."
+            )
+
+        session = sagemaker.Session(default_bucket=self.config.bucket)
+        image_name = deployment.pipeline.extra[ORCHESTRATOR_DOCKER_IMAGE_KEY]
+        execution_role = (
+            self.config.processor_role or sagemaker.get_execution_role()
+        )
+
+        sagemaker_steps = []
+        for step_name, step in deployment.steps.items():
+            command = StepEntrypointConfiguration.get_entrypoint_command()
+            arguments = StepEntrypointConfiguration.get_entrypoint_arguments(
+                step_name=step_name
+            )
+            entrypoint = command + arguments
+
+            step_settings = cast(
+                SagemakerOrchestratorSettings, self.get_settings(step)
+            )
+            execution_role = step_settings.execution_role or execution_role
+
+            processor = sagemaker.processing.Processor(
+                role=execution_role,
+                image_uri=image_name,
+                instance_count=1,
+                sagemaker_session=session,
+                instance_type=step_settings.instance_type,
+                entrypoint=entrypoint,
+                base_job_name=deployment.run_name,
```

Comment:
One other small thing: This `deployment.run_name` looks like `pipeline_name_{date}_{time}`, not sure we should use this here. What is the base job name used for anyway? I guess if it's a prefix that is added to the step name, then using the `orchestrator_run_name` defined below is probably best